File: baekjoon/1260.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#define MAX_SIZE 1001

void dfs(int graph[][MAX_SIZE], int vertex, int *visited, int v) {
	visited[v] = 1;
	printf("%d ", v);
	for (int u = 0; u < vertex + 1; u++) {
		if (graph[v][u] && !visited[u])
			dfs(graph, vertex, visited, u);
	}
}

void bfs(int graph[][MAX_SIZE], int vertex, int *visited, int v) {
	int queue[MAX_SIZE] = {0, };
	int front = -1, rear = -1;
	visited[v] = 1;
	printf("%d ", v);
	queue[++rear] = v;
	while ((rear >= 0) && (front < rear)) {
		v = queue[++front];
		for (int u = 0; u < vertex + 1; u++) {
			if (graph[v][u] && !visited[u]){
				visited[u] = 1;
				printf("%d ", u);
				queue[++rear] = u;
			}
		}
	}
}

int main(void) {
	int vertex, edge, v;
	int graph[MAX_SIZE][MAX_SIZE] = {0, };
	int visited[MAX_SIZE] = {0, };

	scanf("%d %d %d", &vertex, &edge, &v);
	for (int i = 0; i < edge; i++) {
		int a, b;
		scanf("%d %d", &a, &b);
		graph[a][b] = 1;
		graph[b][a] = 1;
	}
	dfs(graph, vertex, visited, v);
	printf("\n");
	memset(visited, 0, sizeof(int) * MAX_SIZE);
	bfs(graph, vertex, visited, v);
	printf("\n");
	return 0;
}

File: baekjoon/1260.py
from sys import stdin
from collections import deque

N, M, V = map(int, stdin.readline().split())

graph = [[] for i in range(N + 1)]
visited = [False] * (N + 1)

for i in range(M):
    a, b = map(int, stdin.readline().split())
    graph[a].append(b)
    graph[b].append(a)

for i in graph:
    i.sort()

def dfs(v):
    visited[v] = True
    print(v, end=' ')
    for i in graph[v]:
        if not visited[i]:
            dfs(i)

def bfs(v):
    q = deque([v])
    visited[v] = True
    print(v, end=' ')
    while q:
        v = q.popleft()
        for i in graph[v]:
            if not visited[i]:
                print(i, end=' ')
                visited[i] = True
                q.append(i)

dfs(V)
print()
visited = [False] * (N + 1)
bfs(V)